sensor/init-tls-certs/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	destinationDir  = "/tmp/certs"
+	legacySourceDir = "/tmp/certs-legacy"
+	newSourceDir    = "/tmp/certs-new"
+)
+
+func main() {
+	log.Print("Started.")
+	realDest, err := sanityCheckDestination()
+	if err != nil {
+		log.Fatalf("Cannot check destination directory %q: %s", destinationDir, err)
+	}
+	log.Printf("Destination directory %q looks sane.", destinationDir)
+
+	files := waitForSource()
+	if err = copyFiles(files, realDest); err != nil {
+		log.Fatalf("Cannot copy files: %s", err)
+	}
+}
+
+func copyFiles(files []string, destDir string) error {
+	for _, file := range files {
+		content, err := os.ReadFile(file)
+		if err != nil {
+			return err
+		}
+		destPath := path.Join(destDir, path.Base(file))
+		perm := os.FileMode(0600) // 0600 is required by Postgres (used by scanner-db)
+		if err = os.WriteFile(destPath, content, perm); err != nil {
+			return err
+		}
+		log.Printf("Copied %q to %q", file, destPath)
+	}
+	return nil
+}
+
+func waitForSource() []string {
+	log.Printf("Looking for files in the source directory %q.", legacySourceDir)
+	for {
+		// Check new certificates first
+		files, err := findFiles(newSourceDir)
+		if err != nil {
+			log.Printf("Error checking certificates in %q: %s", newSourceDir, err)
+		} else {
+			log.Printf("Using new certificates from %q.", newSourceDir)
+			return files
```

Comment:
Let's try to be precise...
```suggestion
			log.Printf("Using %d new certificate files from %q.", len(files), newSourceDir)
			return files
```

Reply:
Done